bundles/EcommerceFrameworkBundle/Model/AbstractOrder.php

```diff
@@ -724,62 +400,30 @@ public function hasDeliveryAddress()
     }
 
     /**
-     * @throws UnsupportedException
-     *
      * @param string $currency
      *
      * @return $this
      */
-    public function setCurrency($currency)
-    {
-        throw new UnsupportedException(__FUNCTION__ . ' is not implemented for ' . get_class($this));
-    }
+    abstract public function setCurrency($currency);
 
     /**
-     * @throws UnsupportedException
-     *
      * @return string
      */
-    public function getCurrency()
-    {
-        throw new UnsupportedException('setPaymentInfo is not implemented for ' . get_class($this));
-    }
+    abstract public function getCurrency();
 
     /**
      * Get voucherTokens - Voucher Tokens
      *
-     * @throws UnsupportedException
-     *
      * @return array
      */
-    public function getVoucherTokens()
-    {
-        throw new UnsupportedException('getVoucherTokens is not implemented for ' . get_class($this));
-    }
+    abstract public function getVoucherTokens();
 
     /**
      * Set voucherTokens - Voucher Tokens
      *
-     * @throws UnsupportedException
-     *
      * @param array $voucherTokens
      *
      * @return \Pimcore\Model\DataObject\OnlineShopOrder
      */
-    public function setVoucherTokens($voucherTokens)
-    {
-        throw new UnsupportedException('setVoucherTokens is not implemented for ' . get_class($this));
-    }
-
-    /**
-     * return cart modification time stamp
-     *
-     * @throws UnsupportedException
-     *
-     * @return int
-     */
-    public function getCartModificationTimestamp()
```

Comment:
please add instead

```
/**
* Get cartHash - Cart Hash
* @return float|null
*/
public function getCartHash ();
```

Reply:
I added it: https://github.com/pimcore/pimcore/pull/7553/commits/528daa63348fe3c960ba06f95b16579b2f7b97b5